awx/main/models/unified_jobs.py

```diff
@@ -810,11 +817,18 @@ def save(self, *args, **kwargs):
 
         # Sanity check: Has the job just completed? If so, mark down its
         # completion time, and record its output to the database.
+        #import sdb
+        #sdb.set_trace()
         if self.status in ('successful', 'failed', 'error', 'canceled') and not self.finished:
-            # Record the `finished` time.
+            # Record the `finished` time and if the job was canceled, set it to null.
             self.finished = now()
+            if self.status == 'canceled':
+                self.finished = None
+                self.canceled = now()
```

Comment:
I don't think `canceled` is a field anymore?

Reply:
oh dang it man, I pushed. This is still a WIP

Reply:
sorry alan, should have closed before I pushed, this was a rebase. still need to clean a lot of stuff up.